py/desispec/zcatalog.py

```diff
@@ -1,15 +1,429 @@
+        
+    Returns
+    -------
+    tab : Astropy Table
+        The final table after adding all the required '*_TARGET' columns
+
+    """
+
+    ## Adding TARGET columns that are not present
+    ## This helps with smooth vstack - if not, it will lead to Masked Columns with missing values
+    ## that can get filled with arbitrary values when saving to a FITS file
+    
+    ## Considering SV1, SV2, SV3 surveys separately
+    ## They have individual TARGET columns
+    sv_surveys = np.array(['sv1', 'sv2', 'sv3'])  
+    
+    if (survey == 'cmx')|(survey.startswith('sv')):
+        ## If survey = cmx|sv1|sv2|sv3 -- the zcatalog does not have 'SCND_TARGET' column
+        col = Column(np.array([0]*len(tab)), name = 'SCND_TARGET', dtype = '>i8')
+        tab.add_column(col)
+        
+    if (survey == 'special')|(survey.startswith('sv')):
+        ## If survey = special|sv1|sv2|sv3 -- the zcatalog does not have 'CMX_TARGET' column
+        col = Column(np.array([0]*len(tab)), name = 'CMX_TARGET', dtype = '>i8')
+        tab.add_column(col)
+        
+    if (survey == 'cmx')|(survey == 'special'):
+        ## If survey = cmx|special -- the zcatalog does not have SV*_TARGET columns
+        for sv in sv_surveys:
+            col1 = Column(np.array([0]*len(tab)), name = f'{sv.upper()}_DESI_TARGET', dtype = '>i8')
+            col2 = Column(np.array([0]*len(tab)), name = f'{sv.upper()}_BGS_TARGET', dtype = '>i8')
+            col3 = Column(np.array([0]*len(tab)), name = f'{sv.upper()}_MWS_TARGET', dtype = '>i8')
+            col4 = Column(np.array([0]*len(tab)), name = f'{sv.upper()}_SCND_TARGET', dtype = '>i8')
+            tab.add_columns([col1, col2, col3, col4])
+            
+    if (survey.startswith('sv')):
+        ## SV1 does not have SV2,SV3* TARGET columns and so on...
+        ## Adding the required columns based on the survey
+        
+        sel = sv_surveys[~(sv_surveys == survey)]
+        for sv in sel:
+            col1 = Column(np.array([0]*len(tab)), name = f'{sv.upper()}_DESI_TARGET', dtype = '>i8')
+            col2 = Column(np.array([0]*len(tab)), name = f'{sv.upper()}_BGS_TARGET', dtype = '>i8')
+            col3 = Column(np.array([0]*len(tab)), name = f'{sv.upper()}_MWS_TARGET', dtype = '>i8')
+            col4 = Column(np.array([0]*len(tab)), name = f'{sv.upper()}_SCND_TARGET', dtype = '>i8')
+            tab.add_columns((col1, col2, col3, col4))
+    
+    return (tab)
+
+####################################################################################################
+####################################################################################################
+
+
+def fix_tables(survey, program, spec_release = 'fuji', version = 'zpix', \
+               all_columns = True, columns_list = None):
+    """
+    This function adds all the required columns and returns the table with all columns, 
+    or pre-selected columns, or with user-given list of columns.
```

Comment:
Style/Clarity: The function name and the comment here imply that `fix_tables` is fixing a pre-existing table, but it appears that it is reading and patching a table from disk.  Consider separating the reading of the table from the "fixing" of the table, or update the function name to include "read" in the name.

If this function still does the reading, it would be better for it to take `filename` as an input instead of re-deriving what it should be (potentially ending up with a different answer from what `glob` found above if we change the file naming scheme in the future).